buildSrc/src/main/kotlin/JpackageExec.kt

```diff
@@ -0,0 +1,313 @@
+import org.gradle.api.DefaultTask
+import org.gradle.api.logging.LogLevel
+import org.gradle.api.provider.HasMultipleValues
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.*
+import org.gradle.internal.os.OperatingSystem
+import org.gradle.kotlin.dsl.listProperty
+import org.gradle.kotlin.dsl.property
+import java.io.FileFilter
+
+/**
+ * Runs the `jpackage` tool (tentatively to be introduced in Java 13) to generate an installer
+ * for the current operating system. `jpackage` cannot generate an installer for a different OS than
+ * the one running the build.
+ *
+ * Because JDK 13 has not yet been released, and because `jpackage` is not yet included in the
+ * early-access builds, a compatible JDK must be downloaded from
+ * [https://jdk.java.net/jpackage/](https://jdk.java.net/jpackage/). The location of the JDK must be
+ * specified with [jdkHome] (eg `/opt/java/jdk-13/`)
+ *
+ */
+open class JpackageExec : DefaultTask() {
+
+    private val objectFactory = project.objects
```

Comment:
Annotate as `@Internal`

Reply:
Since the property is `private`, shouldn't that be unnecessary? There's no public accessor method like there are on no-modifier (public) properties.